IPython/core/debugger.py

```diff
@@ -351,6 +354,17 @@ def hidden_frames(self, stack):
             ip_hide = [h if i > ip_start[0] else True for (i, h) in enumerate(ip_hide)]
         return ip_hide
 
+    def preloop(self):
+        """
+        Save a copy of all frame locals so changes to them are not lost
+        when going up/down in the stack frame.
+        """
+        self.frame_locals = [
+            stack_entry[0].f_locals.copy() for stack_entry in self.stack
```

Comment:
@ccordoba12 , is there any need to copy this dict? 
If there isn't, removing `copy()` could fix the problem mentioned by @MrMino 
```suggestion
            stack_entry[0].f_locals for stack_entry in self.stack
```

Reply:
Could you clarify? What's a `ChainMap`?

Reply:
![image](https://user-images.githubusercontent.com/18216480/132145491-e2167407-1128-40f5-ae82-e7ebacefab23.png)

Reply:
Ok, thanks a lot for the suggestion! I'll take a look at it as soon as Io have some free time.